.github/workflows/rebuildDependencies.yml

```diff
@@ -17,40 +17,61 @@ jobs:
           - platform: msvc-x64
             triplet: x64-windows
             runner: windows-latest
-            # commit before addition of boost filesystem 1.87 that removes Win7 support https://github.com/microsoft/vcpkg/pull/42678
-            commit: ec12d917a85839741f8345905f71b3e7f56d9ddc 
           - platform: msvc-x86
             triplet: x86-windows
             runner: windows-latest
-            # commit before addition of boost filesystem 1.87 that removes Win7 support https://github.com/microsoft/vcpkg/pull/42678
-            commit: ec12d917a85839741f8345905f71b3e7f56d9ddc
           - platform: msvc-arm64
             triplet: arm64-windows
             runner: windows-11-arm
-            # commit before changes to yasm package which seems to break ARM builds https://github.com/microsoft/vcpkg/pull/45856
-            commit: 94a9df6990e426ec3fff1a4ba20016da4aafea70
 
     runs-on: ${{ matrix.runner }}
     defaults:
       run:
         shell: bash
 
     steps:
-      - name: Checkout vcpkg repository
+      # Use latest vcpkg for x64/x86
+      - name: Checkout vcpkg (latest)
         uses: actions/checkout@v4
         with:
           repository: 'microsoft/vcpkg'
-          ref: ${{ matrix.commit }}
+          ref: master
+
+      - name: Prepare overlays (Boost + YASM)
+        run: |
+          mkdir -p vcmi-overlay-boost vcmi-overlay-yasm
+
+          # Clone vcpkg once
+          git clone https://github.com/microsoft/vcpkg.git temp-vcpkg
```

Comment:
you don't need even this clone as the repo was already cloned in the previous step ;)